dash/dash.ino

```diff
@@ -469,22 +486,74 @@ void loop() {
   updateTach(rpm, checkEngineLight);
 }
 
+
+char lat[12];
+char lng[12];
+char speed[8];
+u16 gpsBufOffset = 0;
+
+// each gps frame is 58 bytes long + plus newline and null termination = 60 bytes
+// the gps reads max 10 times per second, or every 100 millis
+// so for at a 1s LoRa message rate, for gpsBuf to hold everything
+// we need 60 * 10 bytes in the buf
+#define GPS_FRAME_SIZE 60
+#define GPS_MAX_SAMPLE_RATE_MILLIS 100
+#define GPS_BUF_SIZE (RADIO_UPDATE_PERIOD_MS / GPS_MAX_SAMPLE_RATE_MILLIS) * GPS_FRAME_SIZE
+char gpsBuf[GPS_BUF_SIZE];
+
+void pushGPSDatum() {
+  if (gps.location.isValid() && gps.date.isValid() && gps.time.isValid())
+  {
+
+    dtostrf(gps.location.lat(), 11, 6, lat);
```

Comment:
arduino `sprintf` doesn't have support for floats, so we need to dump the floats to strings first and then dump those strings into the frame